### PR TITLE
Made image as deliberately optional in v1 Container struct.

### PR DESCRIPTION
**Pull request by gyliu513**
**What this PR does / why we need it**:
Revert https://github.com/kubernetes/kubernetes/pull/47246/
**Which issue this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*: fixes #

**Special notes for your reviewer**:

**Release note**:

```release-note
Mark image as deliberately optional in v1 Container struct.  Many objects in the Kubernetes API inherit the container struct and only Pods require the field to be set.
```

### CODE REVIEW

**Comment by smarterclayton** [comment]
Hold on this, conversation in the other thread.

**Comment by smarterclayton** [comment]
Marking this a candidate, still needs discussion.  Not required for feature freeze but don't want to forget it.

**Comment by bgrant0607** [comment]
@smarterclayton Which other thread? #47246? Indeed, a user there reported that this broke them, since validation runs in kubectl. I think we need to revert the change and backport it to 1.7. I'd even like to remove the validation from the controllers in v1beta2 in 1.9 and just have them not create pods if the field is empty.

**Comment by smarterclayton** [comment]
Agreed, revert and backport.  Just qualifies as bug.

**Comment by smarterclayton** [comment]
Can you rebase so we can get this merged and back ported?

**Comment by gyliu513** [comment]
I was trying to rebase, but encounter an issue of `./hack/update-all.sh` failed here https://github.com/kubernetes/kubernetes/issues/52632 , checking what is wrong with it.

File: staging/src/k8s.io/api/core/v1/types.go [hunk 1]
```diff
@@ -1887,7 +1887,11 @@ type Container struct {
 	Name string `json:"name" protobuf:"bytes,1,opt,name=name"`
 	// Docker image name.
 	// More info: https://kubernetes.io/docs/concepts/containers/images
-	Image string `json:"image" protobuf:"bytes,2,opt,name=image"`
+	// This field was intentionally optional, refer to the following for more info.
```
Comment (smarterclayton) [inline staging/src/k8s.io/api/core/v1/types.go]: The message here should probably be:

```
This field is optional to allow higher level config management to default or override container images in workload controllers like Deployments and StatefulSets.
```

We generally would not link to issues in our comments.

**Comment by smarterclayton** [comment]
Marking for 1.8 - this was a regression in API behavior and we should not ship another release without it.

File: staging/src/k8s.io/api/core/v1/types.go (hunk 1 above)
Comment (gyliu513) [reply staging/src/k8s.io/api/core/v1/types.go]: Done

**Comment by dims** [comment]
@gyliu513 - Can you please see the failures and help get the tests go green? i am afraid this will miss the 1.8 otherwise.

**Comment by gyliu513** [comment]
@dims I filed an issue here https://github.com/kubernetes/kubernetes/issues/52727 reporting some errors when `hack/update-all.sh`, can you help check? I suspect the error may cause some files can not be generated.

**Comment by gyliu513** [comment]
@smarterclayton @dims

**Comment by spiffxp** [comment]
@smarterclayton this needs an `/approve no-issue` or @gyliu513 needs to refer to an issue (not PR) in the "fixes #" part of the PR description before the bot will add an `approved` label

**Pull request merged**